crates/hotshot/src/lib.rs

```diff
@@ -641,14 +651,24 @@ impl<TYPES: NodeType> HotShotInitializer<TYPES> {
         Ok(Self {
             inner: Leaf::genesis(instance_state),
             instance_state: instance_state.clone(),
+            start_view: TYPES::Time::new(0),
         })
     }
 
-    /// reload previous state based on most recent leaf and the instance-level state.
-    pub fn from_reload(anchor_leaf: Leaf<TYPES>, instance_state: TYPES::InstanceState) -> Self {
+    /// Reload previous state based on most recent leaf and the instance-level state.
+    ///
+    /// # Arguments
+    /// *  `start_view` - The minimum view number that we are confident won't lead to a double vote
+    /// after restart.
+    pub fn from_reload(
+        anchor_leaf: Leaf<TYPES>,
+        instance_state: TYPES::InstanceState,
+        start_view: TYPES::Time,
```

Comment:
Would we ever set `start_view` to anything but `anchor_leaf.get_view_number()` or `anchor_leaf.get_view_number() + 1`?

Reply:
Yes. Suppose anchor_leaf has view 100, but when your node shut down the latest view was 200: there were many views where you voted but never reached decide. If you restart in view 100 or 101, you may end up voting again in views 101-200, while there were already votes with your signature floating around for those views. This can end up with your node getting slashed.

So I think it is important that consensus works even if `start_view` is much greater than `anchor_leaf.get_view_number()` (as long as a quorum of nodes agree that nothing has been committed between `anchor_leaf` and `start_view`